### ISSUE TITLE
[storage] When azure storage call fails with UnexpectedStatusCodeError, no detailed error could be accessed

### ISSUE THREAD

**Issue by yuwaMSFT2** [Azure/azure-sdk-for-go#891]
When using the storage SDK to send request to Azure Storage, if it fails with unexpected status, the SDK returns an UnexpectedStatusCodeError, which only has the returned status code. Actually the response body has the detailed information of the failure, including the x-ms-request-id. 

This is important since we need these information to either correlate the failed request in storage log, or open incident to the Azure Storage team (needs the request Id for them to investigate).

A simple approach would be to include the response in the error (we did this as a hack to work around):

```go
type UnexpectedStatusCodeError struct {
	allowed  []int
	got      int
	response interface{}
}
func (e UnexpectedStatusCodeError) Error() string {
return fmt.Sprintf(
	"storage: status code from service response is %s; was expecting %s. Response: %+v", 
	got,
	strings.Join(expected, " or "),
	e.response)
}
```

And in checkRespCode, pass in the response:
`func checkRespCode(resp interface{}, allowed []int) error`

**Comment by jhendrixMSFT** [comment]
Note that we are slowly deprecating the current storage package.  If you're using blob storage take a look at the new blob storage APIs [here](https://github.com/Azure/azure-storage-blob-go).  We're working on getting this package included in the azure-sdk-for-go so you don't have to get/import it separately.

**Comment by jhendrixMSFT** [comment]
@yuwaMSFT2 Please note that we're deprecating the blob storage functionality in the storage package.  The preferred way to interact with blob storage is the new azblob package [here](https://github.com/Azure/azure-storage-blob-go).  We are also working on rolling up that package into this repo.

**Comment by jhendrixMSFT** [comment]
Fixed in v14.0.0.
